### PR TITLE
implement binary_boolean for chunked encoding

### LINKED ISSUE #1443: Implement binary_boolean_fn for ChunkedArray

### PR DESCRIPTION
**Pull request by doki23**
Close #1443

### CODE REVIEW

**Comment by doki23** [comment]
@robert3005 @gatesn Would you please help reivew?

**Comment by gatesn** [comment]
Sorry, @doki23  - we haven't had an external contribution that required us to run benchmarks before. Just trying to figure out how to get these to run with the right secrets / permissions etc! Please bear with us

File: vortex-array/src/array/chunked/compute/boolean.rs [hunk 1]
```diff
@@ -14,21 +14,53 @@ impl BinaryBooleanFn<ChunkedArray> for ChunkedEncoding {
...
             let sliced = slice(rhs, idx, idx + chunk.len())?;
-            let result = match op {
-                BinaryOperator::And => and(&chunk, &sliced),
-                BinaryOperator::AndKleene => and_kleene(&chunk, &sliced),
-                BinaryOperator::Or => or(&chunk, &sliced),
-                BinaryOperator::OrKleene => or_kleene(&chunk, &sliced),
-            };
-            chunks.push(result?);
+            let result = binary_boolean(&chunk, &sliced, op)?;
+            nullability = result.dtype().nullability();
```
Comment (robert3005) [inline vortex-array/src/array/chunked/compute/boolean.rs]: I think this is not quite correct - if the any of the chunks is nullable the chunked array should be nullable. We can move this out of the loop and add

```
        let nullable = chunks.iter().any(|c| c.dtype().is_nullable());
```

**Review by robert3005 — DISMISSED** [review]
Left one comment regarding nullability

File: vortex-array/src/array/chunked/compute/boolean.rs (hunk 1 above)
Comment (gatesn) [reply vortex-array/src/array/chunked/compute/boolean.rs]: All chunks in a chunked array must have the same DType

Comment (robert3005) [reply vortex-array/src/array/chunked/compute/boolean.rs]: I forgot that part, then we should be able to take the first chunk's dtype

Comment (gatesn) [reply vortex-array/src/array/chunked/compute/boolean.rs]: No, nullability should be `lhs.dtype().is_nullable() || rhs.dtype().is_nullable()`

Comment (robert3005) [reply vortex-array/src/array/chunked/compute/boolean.rs]: I don’t think this works this way for all the Boolean logic functions.

Comment (doki23) [reply vortex-array/src/array/chunked/compute/boolean.rs]: My understanding is that the data type (dtype) of the resulting chunked array matches that of its children. These children consist of the outcomes from each `binary_boolean` call, indicating that the dtype of the first chunk is relevant. Perhaps we could clone the dtype of the first chunk like
```rust
let dtype = chunks
    .first()
    .map(|chunk| chunk.dtype().clone())
    .unwrap_or(DType::Bool(Nullability::Nullable));
Ok(Some(ChunkedArray::try_new(chunks, dtype)?.into_array()))
```

Comment (robert3005) [reply vortex-array/src/array/chunked/compute/boolean.rs]: I think what @gatesn suggested is the right answer. There might not be any chunks to take nullability from

Comment (doki23) [reply vortex-array/src/array/chunked/compute/boolean.rs]: I agree. It's same with arrow_boolean.

**Review by gatesn — APPROVED** [review]

**Pull request merged**